### PR TITLE
[rv_plic] RV_PLIC Updates to conform to the spec

### PR DESCRIPTION
**Pull request by msfschaffner**
Formal change request can be found here: #7670.

---------------------------------------------------
This PR aligns a couple of RV_PLIC opens (#1491, #7435):
- The [PLIC spec](https://github.com/riscv/riscv-plic-spec/blob/master/riscv-plic.adoc) does not say anything about edge/level triggered interrupts and does not provide any CSRs to configure this. I think in theory, this means that both can be supported by a specific hardware implementation. However since that is a hardware property, I don't think SW needs to be able to configure this via a CSR. Since our comportability spec clearly says OpenTitan IP uses [positive level triggered interrupts](https://docs.opentitan.org/doc/rm/comportability_specification/#interrupt-handling), I have removed the associated CSR (and related DIF functions) from the RV_PLIC implementation.
- Our RV_PLIC implementation so far did not synchronize the interrupt signals to the PLIC clock domain (#7435). The second commit takes care of that.
- The third commit aligns the address map with the [PLIC spec](https://github.com/riscv/riscv-plic-spec/blob/master/riscv-plic.adoc) (see also #1491). This introduces a large 64MB address range with mostly reserved addresses, which is a bit unfortunate. The PLIC is therefore moved to a different offset in the [Earlgrey memory map](https://docs.google.com/spreadsheets/d/1ptfdrG5RWiYHKtjYyDO1ubGYPPlN_4YQfzRKO1G1Mls/edit#gid=0) to accommodate this large chunk of reserved addresses.
- Note that the MSIP registers are technically not part of the PLIC, but the CLIC as far as I understand (see [SiFive U54-MC ](https://static.dev.sifive.com/U54-MC-RVCoreIP.pdf) as an example). For the PLIC to be compliant, we need to move the MSIP registers to an offset that does not lie within the standardized PLIC address range.

Could someone double check the address map to make sure we get this right?

Signed-off-by: Michael Schaffner <msf@opentitan.org>

### CODE REVIEW

**Comment by eunchan** [comment]
I've discussed with Michael offline. This looks good overall. I added edge/ level detection in RV_PLIC to catch the ghost interrupt and to support 3rd party IP. But I agree we won't see 3p IP in near future. Also, we can mandate the IP to use prim interrupt cell to ensure the interrupt is always level.

**Comment by eunchan** [comment]
CC @cindychip to check the coverage again.

File: util/tlgen/validate.py [hunk 1]
```diff
@@ -282,6 +287,13 @@ def validate(obj: OrderedDict) -> Xbar:  # OrderedDict -> Xbar
 
                 addr_entry = (address_from, address_to)
 
+                if checkBaseSizeOverlap(address_from, size):
+                    log.error(
+                        "Size mask and base address are overlapping. "
+                        " Check the config. Addr(0x%x - 0x%x)"
+                        % (addr_entry[0], addr_entry[1]))
+                    raise SystemExit("Base/size overlapping error occurred")
```
Comment (msfschaffner) [inline util/tlgen/validate.py:295]: @weicaiyang added this check to avoid running into this address decoding issue again.

**Comment by msfschaffner** [comment]
@cindychip the latest force push includes some fixes for the FPV environment (add synchronizer delays, remove edge-triggered option). there are still some failing autogenerated CSR assertions which could be unrelated to this change (I did not look too closely, tough).

**Comment by msfschaffner** [comment]
Rebased on latest changes on `master`.

**Review by cindychip — COMMENTED** [review]
The FPV part looks good to me. I will check the coverage and fix the CSR autogen issue.

**Comment by cindychip** [comment]
Thanks again for this PR Michael, the csr assertion FPV error is due to the large array of expected values from the CSR registers, that FPV automatically black-box it. I will make some change in the script to modify that.
The assertion coverage looks good when the CSR values are excluded, and seems to still satisfy V2 status. I will provide a coverage report once fixed the blackbox issue.

**Comment by msfschaffner** [comment]
Great, thanks @cindychip!

**Review by cindychip — APPROVED** [review]
Here are the coverage data:
V2 requires code coverage above 90% --> This new model is 99.56%
V2 requires COI coverage above 75%   --> This new model is 75.08% (The main issue are some deadcode in `rv_plic_gateway` and in `prim_alert`).
But all the metrics still meet V2 requirement.

Screenshot from 2021-09-03 17-05-56

**Comment by msfschaffner** [comment]
Thanks for all the feedback! I am merging this now.

**Pull request merged**